### PR TITLE
Use a nice mpath-WWID serial for multipath devices.

### PR DESCRIPTION
**Pull request by xnox**

### CODE REVIEW

**Comment by xnox** [comment]
Or maybe this should be done in subiquity UX instead? Because with this in place, the following curtin config got generated:

```
storage:
  config:
  - {id: disk-0, type: disk, ptable: gpt, serial: mpath-36005076306ffd6b60000000000002404,
    path: /dev/dm-0, wipe: superblock, preserve: false, name: '', grub_device: true}
  - {id: part-0, type: partition, device: disk-0, size: 1048576, flag: bios_grub,
    preserve: false}
  - {id: part-1, type: partition, device: disk-0, size: 10734272512, flag: '', preserve: false}
  - {id: fs-0, type: format, fstype: ext4, volume: part-1, preserve: false}
  - {id: mount-0, type: mount, device: fs-0, path: /}
```

And i'm not sure if curtin will get confused by such a "serial".

File: probert/storage.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ def model(self):
 
     @property
     def serial(self):
-        return self._get_hwvalues(['ID_SERIAL', 'ID_SERIAL_SHORT'])
+        return self._get_hwvalues(['ID_SERIAL', 'ID_SERIAL_SHORT', 'DM_UUID'])
```
Comment (raharper) [inline probert/storage.py]: DM_UUID doesn't imply multipath, rather a disk managed by device-mapper.  lvm, raid , crypt may all have a DM_UUID.

I don't think we should equate DM_UUID as a "disk" serial. I think we should have dm property that indicate what sort of disk it is (lvm, raid), and I think multipath can be detected via multipath tools (or implicitly via duplicate serial numbers).

Comment (xnox) [reply probert/storage.py]: True.

It's just in subiquity UI, `/dev/dm-0` is not descriptive at all, as to what it is. And I want to display a WWN somehow, which for multipath devices is available in multiple ways, one of which is DM_UUID. However subiquity UI can get to, and display DM_UUID without requiring it to be in the serial field. Currently all other DM devices are filtered out on the subiquity side of probe(), but that may change in the future.

**Comment by mwhudson** [comment]
I think I'd rather make this a more explicit thing in the UI rather than pretending the DM_UUID is a serial number, i.e. change subiquity.models.filesystem.Disk.label rather than this.

File: probert/storage.py (hunk 1 above)
Comment (raharper) [reply probert/storage.py]: FWIW, curtin can use mpath-$ID_SERIAL just fine:

>>> from curtin import block
>>> import glob, os
>>> glob.glob('/dev/disk/by-id/*mpath-36005076306ffd6b60000000000002406*')
>>> os.path.realpath(glob.glob('/dev/disk/by-id/*mpath-36005076306ffd6b60000000000002406*')[0])
'/dev/dm-2'
>>> block.lookup_disk('mpath-36005076306ffd6b60000000000002406')
'/dev/dm-2'

Comment (xnox) [reply probert/storage.py]: ack.

**Comment by mwhudson** [comment]
Closing in favour of changing subiquity then, feel free to reopen if that's not the right thing to do...

**Pull request closed without merging**